src/libFLAC/stream_encoder_framing.c

```diff
@@ -96,9 +96,13 @@ FLAC__bool FLAC__add_metadata_block(const FLAC__StreamMetadata *metadata, FLAC__
 			FLAC__ASSERT(metadata->data.stream_info.bits_per_sample <= (1u << FLAC__STREAM_METADATA_STREAMINFO_BITS_PER_SAMPLE_LEN));
 			if(!FLAC__bitwriter_write_raw_uint32(bw, metadata->data.stream_info.bits_per_sample-1, FLAC__STREAM_METADATA_STREAMINFO_BITS_PER_SAMPLE_LEN))
 				return false;
-			FLAC__ASSERT(metadata->data.stream_info.total_samples < (FLAC__U64L(1) << FLAC__STREAM_METADATA_STREAMINFO_TOTAL_SAMPLES_LEN));
-			if(!FLAC__bitwriter_write_raw_uint64(bw, metadata->data.stream_info.total_samples, FLAC__STREAM_METADATA_STREAMINFO_TOTAL_SAMPLES_LEN))
-				return false;
+			if(metadata->data.stream_info.total_samples > (FLAC__U64L(1) << FLAC__STREAM_METADATA_STREAMINFO_TOTAL_SAMPLES_LEN)){
```

Comment:
Shouldn't this be `>=' ?

Reply:
From an overflow point-of-view, yes, this should be >=, and I didn't notice that. Now I think of it however, functionally, it should make no difference, because `metadata->data.stream_info.total_samples = (FLAC__U64L(1) << FLAC__STREAM_METADATA_STREAMINFO_TOTAL_SAMPLES_LEN)` overflows to 0, which is the value that we are setting anyway.

I could fix this and force-push a different commit. However, I could also leave it as-is, and check for it in a test (as you suggested). If something changes in the overflow behaviour, which is something undesirable by itself, this could be picked up by the test.

Please tell me which behaviour would be preferred, I'm not really sure.

Concerning writing a test, I will add a commit to this branch today to add this to test_flac.sh. Running the test takes quite a while and uses 54MiB of disk space, but it is doable. I will place it under FLAC__TEST_LEVEL=2 to not stress the CI too much.

Reply:
```suggestion
			if(metadata->data.stream_info.total_samples >= (FLAC__U64L(1) << FLAC__STREAM_METADATA_STREAMINFO_TOTAL_SAMPLES_LEN)){
```